src/client/components/Products.tsx

```diff
@@ -1,18 +1,40 @@
-export default function Products() {
+import React from "react";
+import styles from "../styles/Product.module.css";
+import IPFSDownload from './IpfsDownload';
+
+type Props = {
+  product: {
+    id: number;
+    name: string;
+    price: string;
+    description:string;
+    image_url:string;
+    filename:string;
+    hash:string;
+  };
+}
+
+export default function Product({ product }: Props) {
```

Comment:
You can declare types for functions like this, but for react functional components it'll be nice to extend the `FC` interface from react

Example
``` javascript
import { FC } from 'react'

interface IProp {
  person: {
    name: string
  }
}

const Test: FC<IProp> = ({ person }) => {
  return <h1>This is a basic text {person.name}</h1>
}

export default Test
```